grpc-circuitbreaker-utils/src/main/java/org/hypertrace/circuitbreaker/grpcutils/resilience/ResilienceCircuitBreakerProvider.java

```diff
@@ -0,0 +1,99 @@
+package org.hypertrace.circuitbreaker.grpcutils.resilience;
+
+import io.github.resilience4j.circuitbreaker.CircuitBreaker;
+import io.github.resilience4j.circuitbreaker.CircuitBreakerConfig;
+import io.github.resilience4j.circuitbreaker.CircuitBreakerRegistry;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.ConcurrentHashMap;
+import lombok.extern.slf4j.Slf4j;
+import org.hypertrace.circuitbreaker.grpcutils.CircuitBreakerConfigParser;
+
+/** Utility class to provide Resilience4j CircuitBreaker */
+@Slf4j
+class ResilienceCircuitBreakerProvider {
+
+  private static final String SHARED_KEY = "SHARED_KEY";
+  private final CircuitBreakerRegistry circuitBreakerRegistry;
+  private final Map<String, CircuitBreakerConfig> circuitBreakerConfigMap;
+  private final Map<String, CircuitBreaker> circuitBreakerCache = new ConcurrentHashMap<>();
```

Comment:
Are there any concerns about this growing too large? If we were to use a bad key (e.g. request ID) as our circuit breaker it would grow unbounded. Consider if we should use a true simple cache here like below to prevent a bad config from crashing a service.
```
        CacheBuilder.newBuilder()
            .expireAfterAccess(Duration.ofHours(1)) // or from config
            .maximumSize(10_000)
            .build(CacheLoader.from(this::buildCircuitBreaker));
```

Reply:
Updated